docs/libraries/typespec-client-generator-core/reference/linter.md

```diff
@@ -0,0 +1,31 @@
+---
+title: "Linter usage"
+toc_min_heading_level: 2
+toc_max_heading_level: 3
+---
+
+# Linter
+
+## Usage
+
+Add the following in `tspconfig.yaml`:
+
+```yaml
+linter:
+  extends:
+    - "@azure-tools/typespec-client-generator-core/all"
+```
+
+## RuleSets
+
+Available ruleSets:
+
+- `@azure-tools/typespec-client-generator-core/all`
+- `@azure-tools/typespec-client-generator-core/best-practices:csharp`
+
+## Rules
+
+| Name                                                                 | Description                                                             |
+| -------------------------------------------------------------------- | ----------------------------------------------------------------------- |
+| `@azure-tools/typespec-client-generator-core/require-client-suffix`  | Client names should end with 'Client'.                                  |
```

Comment:
can we specify which ruleset each of these are in?

Reply:
That file is auto generated with the list so would need to update the ref doc generator. If you have some idea on what'd you like to see feel free to suggest a new style but this should work for a generic linter